src/modules/rlm_lua/lua.c

```diff
@@ -782,11 +838,32 @@ int do_lua(rlm_lua_thread_t *thread, REQUEST *request, char const *funcname)
 		goto error;
 	}
 
-	if (lua_pcall(L, 0, 0, 0) != 0) {
+	if (lua_pcall(L, 0, 1, 0) != 0) {
 		char const *msg = lua_tostring(L, -1);
 		REDEBUG("Call to %s failed: %s", funcname, msg ? msg : "unknown error");
 		goto error;
 	}
 
-	return 0;
+	
+	/*
+	 *	the functions should return
+	 */
+	if (lua_isnil(L, -1)) goto error;
+
+	ret = lua_tointeger(L, -1);
```

Comment:
We don't know it's an integer yet.  We need to check the type first, and error out if it's not what we expect.  If you wanted to be nice and found the return value was a string, you could just look it up in the rcode_table and use that if it matched an entry.

e.g. return "ok"